### PR TITLE
Bump detector semconv to v1.24.0

### LINKED ISSUE #4698: Upgrade all resource detectors to use go.opentelemetry.io/otel/semconv/v1.24.0
Related issue: https://github.com/open-telemetry/opentelemetry-go/issues/4753

### PR DESCRIPTION
**Pull request by MrAlias**
Fix #4698 

- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/ecs` are upgraded to v1.24.0.
- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/lambda` are upgraded to v1.24.0.
- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/ec2` are upgraded to v1.24.0.
- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/eks` are upgraded to v1.24.0.
- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/gcp` are upgraded to v1.24.0.
- The semantic conventions used by `go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda/test` are upgraded to v1.24.0.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The semantic conventions used by `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/example` are upgraded to v1.20.0. (#4320)
 - The semantic conventions used by `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`are upgraded to v1.20.0. (#4320)
 - Updated configuration schema to include `schema_url` for resource definition and `without_type_suffix` and `without_units` for the Prometheus exporter. (#4727)
+- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/ecs` are upgraded to v1.24.0. (#4803)
+- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/lambda`.are upgraded to v1.24.0. (#4803)
+- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/ec2`.are upgraded to v1.24.0. (#4803)
+- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/eks`.are upgraded to v1.24.0. (#4803)
+- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/gcp`.are upgraded to v1.24.0. (#4803)
+- The semantic conventions used by `go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda/test`.are upgraded to v1.24.0. (#4803)
```
Comment (carrbs) [inline CHANGELOG.md:33]: You got some extra "`.`"s (i.e. "```/foo/bar`.are upgraded...```")

**Review by carrbs — CHANGES_REQUESTED** [review]
Do we need to update `semconv` here too?
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/3cecdcf33c4701a3abd8557d38ffde090a23b696/instrumentation/github.com/aws/aws-lambda-go/otellambda/lambda.go#L27

File: CHANGELOG.md (hunk 1 above)
Comment (pellared) [reply CHANGELOG.md]: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4803/commits/35be77a1d911041a3009387c1578330402d25631

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,12 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The semantic conventions used by `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/example` are upgraded to v1.20.0. (#4320)
 - The semantic conventions used by `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`are upgraded to v1.20.0. (#4320)
 - Updated configuration schema to include `schema_url` for resource definition and `without_type_suffix` and `without_units` for the Prometheus exporter. (#4727)
+- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/ecs` are upgraded to v1.24.0. (#4803)
+- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/lambda` are upgraded to v1.24.0. (#4803)
```
Comment (pellared) [inline CHANGELOG.md:29]:
```suggestion
- The semantic conventions used by `go.opentelemetry.io/contrib/detectors/aws/lambda` resource detector are upgraded to v1.24.0. (#4803)
```

OR

https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4803#pullrequestreview-1819478769

Personally I am leaning more towards adding `resource detector` to all changelog entries.

**Review by dashpole — APPROVED** [review]

**Comment by MrAlias** [comment]
> Do we need to update `semconv` here too?
> 
> https://github.com/open-telemetry/opentelemetry-go-contrib/blob/3cecdcf33c4701a3abd8557d38ffde090a23b696/instrumentation/github.com/aws/aws-lambda-go/otellambda/lambda.go#L27

Eventually yes, but it is not scoped within this PR.

**Review by dmathieu — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Pull request merged**